src/jsp/com/cubrid/jsp/data/CUBRIDPacker.java

```diff
@@ -0,0 +1,237 @@
+ *   may be used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+ * OF SUCH DAMAGE.
+ *
+ */
+
+package com.cubrid.jsp.data;
+
+import cubrid.jdbc.driver.CUBRIDResultSet;
+import cubrid.sql.CUBRIDOID;
+import java.io.UnsupportedEncodingException;
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.sql.ResultSet;
+import java.text.SimpleDateFormat;
+
+public class CUBRIDPacker {
+    private ByteBuffer buffer;
+
+    public CUBRIDPacker(ByteBuffer buffer) {
+        this.buffer = buffer;
+    }
+
+    public CUBRIDPacker(byte[] byteArray) {
+        this(ByteBuffer.wrap(byteArray));
+    }
+
+    public void setBuffer(ByteBuffer buffer) {
+        this.buffer = buffer;
+    }
+
+    public ByteBuffer getBuffer() {
+        return this.buffer;
+    }
+
+    public void packInt(int value) {
+        align(DataUtilities.INT_ALIGNMENT);
+        buffer.putInt(value);
+    }
+
+    public void packBool(boolean value) {
+        buffer.putInt(value ? 1 : 0);
+    }
+
+    public void packShort(short value) {
+        align(DataUtilities.SHORT_ALIGNMENT);
+        buffer.putShort(value);
+    }
+
+    public void packBigInt(long value) {
+        align(DataUtilities.MAX_ALIGNMENT);
+        buffer.putLong(value);
+    }
+
+    public void packFloat(float value) {
+        // TODO: alignment is not considered yet in cubpacking::packer
+        // align(DataUtilities.FLOAT_ALIGNMENT);
+        buffer.putFloat(value);
+    }
+
+    public void packDouble(double value) {
+        // TODO: alignment is not considered yet in cubpacking::packer
+        // align(DataUtilities.DOUBLE_ALIGNMENT);
+        buffer.putDouble(value);
+    }
+
+    public void packString(String value) {
+        packCString(value.getBytes());
+    }
+
+    public void packString(String value, String charset) throws UnsupportedEncodingException {
+        packCString(value.getBytes(charset));
+    }
+
+    public void packCString(byte[] value) {
+        int len = value.length;
+        if (len < DataUtilities.MAX_SMALL_STRING_SIZE) {
+            ensureSpace(value.length + 1); // str + len
```

Comment:
Is the len information resolved to 1 byte? Maybe 4 bytes are needed?
ensureSpace(value.length + 4);